hw/ip/otbn/dv/rig/rig/gens/edge_load_store.py

```diff
@@ -0,0 +1,234 @@
+                self.bnlid.operands[2].op_type.signed and
+                isinstance(self.bnlid.operands[3].op_type, OptionOperandType) and
+                isinstance(self.bnlid.operands[4].op_type, OptionOperandType))):
+            raise RuntimeError('BN.LID instruction from instructions file is'
+                               'not the shape expected by MisalignedBadLoadStore'
+                               'generator.')
+
+        # BN.SID Instruction Checks
+
+        self.bnsid = self._get_named_insn(insns_file, 'bn.sid')
+
+        # bn.sid expects the operands: grs1, grs2, offset, grs1_inc,
+        # grs2_inc
+        if len(self.bnsid.operands) != 5:
+            raise RuntimeError('Unexpected number of operands for bn.sid')
+
+        if not (isinstance(self.bnsid.operands[0].op_type, RegOperandType) and
+                self.bnsid.operands[0].op_type.reg_type == 'gpr' and
+                not self.bnsid.operands[0].op_type.is_dest() and
+                isinstance(self.bnsid.operands[1].op_type, RegOperandType) and
+                self.bnsid.operands[1].op_type.reg_type == 'gpr' and
+                not self.bnsid.operands[1].op_type.is_dest() and
+                isinstance(self.bnsid.operands[2].op_type, ImmOperandType) and
+                self.bnsid.operands[2].op_type.signed and
+                isinstance(self.bnsid.operands[3].op_type, OptionOperandType) and
+                isinstance(self.bnsid.operands[4].op_type, OptionOperandType)):
+            raise RuntimeError('BN.SID instruction from instructions file is'
+                               'not the shape expected by MisalignedBadLoadStore'
+                               'generator.')
+
+        for insn in insns_file.insns:
+            if (insn.mnemonic in ['lw', 'sw', 'bn.lid', 'bn.sid']):
+                weight = cfg.insn_weights.get(insn.mnemonic)
+                if weight > 0:
+                    self.weights.append(weight)
+                    self.insns.append(insn)
+
+        # Check that at least one instruction has a positive weight
+        assert len(self.insns) == len(self.weights)
+        if not self.weights:
+            raise ValueError('Config at {} defines a zero weight '
+                             'for all instructions.'
+                             .format(cfg.path))
+
+    def gen(self,
+            cont: GenCont,
+            model: Model,
+            program: Program) -> Optional[GenRet]:
+        # Return None if this is the last instruction in the current gap
+        # because we need to either jump or do an ECALL to avoid getting stuck.
+        if program.get_insn_space_at(model.pc) <= 1:
+            return None
+
+        # Determine the offset range for XW
+        offset_rng = self.imm_op_type.get_op_val_range(model.pc)
+        assert offset_rng is not None
+
+        # Max/Min Offsets for XW (-2048, 2047)
+        max_offset = offset_rng[1]
+        min_offset = offset_rng[0]
```

Comment:
Probably cleaner to write something like `min_offset, max_offset = offset_rng`. Similarly for the BN instruction below.